src/controllers/media.ts

```diff
@@ -276,60 +291,71 @@ export const getBySanitizedTitleV2 = async(ctx: ParameterizedContext): Promise<M
-  const { year }: UmsQueryParams = ctx.query;
-  if (!dirOrFilename) {
+  const { title, year }: UmsQueryParams = ctx.query;
+  if (!title) {
     throw new ValidationError('title is required');
   }
 
-  let dbMeta: SeriesMetadataInterface = await SeriesMetadata.findSimilarSeries(dirOrFilename, year);
-  if (dbMeta) {
-    return ctx.body = dbMeta;
+  const dbMeta = await externalAPIHelper.getSeriesMetadata(null, title, year);
+  return ctx.body = dbMeta;
+};
+
+/**
+ * Gets season information from TMDB since it's the only API
+ * we use that has that functionality.
+ *
+ * @param ctx 
+ * @returns 
+ */
+export const getSeason = async(ctx: ParameterizedContext): Promise<SeasonMetadataInterface> => {
+  const { season, title, year }: UmsQueryParams = ctx.query;
+  if (!title || !season) {
+    throw new ValidationError('title and season are required');
   }
+  const seasonNumber = Number(season);
 
-  const failedLookupQuery: FailedLookupsInterface = { title: dirOrFilename, type: 'series' };
+  // Return early for previously-failed lookups
+  const failedLookupQuery: FailedLookupsInterface = { title, season, type: 'season' };
   if (year) {
     failedLookupQuery.year = year;
   }
-
   if (await FailedLookups.findOne(failedLookupQuery, '_id', { lean: true }).exec()) {
     await FailedLookups.updateOne(failedLookupQuery, { $inc: { count: 1 } }).exec();
     throw new MediaNotFoundError();
   }
 
-  const parsed = episodeParser(dirOrFilename);
-  dirOrFilename = parsed && parsed.show ? parsed.show : dirOrFilename;
-  const searchRequest: SearchRequest = {
-    name: dirOrFilename,
-    reqtype: 'series',
-  };
-  if (year) {
-    searchRequest.year = Number(year);
+  const seriesMetadata = await externalAPIHelper.getSeriesMetadata(null, title, year);
+  if (!seriesMetadata?.tmdbID) {
+    await FailedLookups.updateOne(failedLookupQuery, { $inc: { count: 1 } }, { upsert: true, setDefaultsOnInsert: true }).exec();
+    throw new MediaNotFoundError();
   }
-  const tvSeriesInfo = await imdbAPI.get(searchRequest);
 
-  if (!tvSeriesInfo) {
-    if (year) {
-      /**
-       * If the client specified a year, it may have been incorrect because of
-       * the way filename parsing works; the filename Galactica.1980.S01E01 might
-       * be about a series called "Galactica 1980", or a series called "Galactica"
-       * from 1980.
-       * So, we attempt the lookup again with the year appended to the title.
-       */
-      ctx.query.title = ctx.query.title + ' ' + year;
-      ctx.query.year = '';
-      return getSeriesByTitle(ctx);
-    }
+  // Start TMDB lookups
+  const seasonRequest = {
+    // eslint-disable-next-line @typescript-eslint/camelcase
+    append_to_response: 'images,external_ids,credits',
+    id: seriesMetadata.tmdbID,
+    // eslint-disable-next-line @typescript-eslint/camelcase
+    season_number: seasonNumber,
+  };
+
+  const tmdbResponse = await moviedb.seasonInfo(seasonRequest);
```

Comment:
From what I can tell tracing the code it seems fairly likely that we might not have the `tmdbID` stored against the season metadata if it had previously been found and saved from OMDB. Is that correct/ could cause this stuff not to work?

Reply:
TMDB is the only API that serves season metadata, so there will always be a TMDB from those results